scripts/compress_images_test.py

```
@@ -0,0 +1,250 @@
+        self.assertEqual(compression_type, 'LZW')
+
+    def test_find_compressible_images(self) -> None:
+        """Test find_compressible_images method."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        result_images: List[CompressedImageInfo] = (
+            compressor.find_compressible_images()
+        )
+
+        self.assertEqual(len(result_images), 1)
+
+        for image in result_images:
+            self.assertIn(image['path'].suffix, ['.png', '.jpg', '.webp'])
+            self.assertTrue(image['original_size'] > image['new_size'])
+
+    def test_compress_images(self) -> None:
+        """Test image compression process."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        result_images: List[CompressedImageInfo] = (
+            compressor.find_compressible_images()
+        )
+
+        self.assertGreaterEqual(len(result_images), 1)
+        compressor.compress_images(result_images)
+        for image in result_images:
+            original_path = image['path']
+            original_size = image['original_size']
+
+            rel_path = os.path.relpath(original_path, self.temp_dir)
+            output_path = pathlib.Path(self.output_dir) / rel_path
+
+            self.assertTrue(output_path.exists())
+            compressed_size = output_path.stat().st_size
+            self.assertLess(compressed_size, original_size)
+
+    def test_compress_images_directory_removal_error(self) -> None:
+        """Test error handling when removing output directory fails."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        os.makedirs(self.output_dir, exist_ok=True)
+        mock_rmtree = mock.MagicMock(ignore_errors=OSError('Permission denied'))
```

Why are you ignoring this error here?

---

I wrote this ignore_error when I didn't marked it false over main script, and forgot to remove error. But updated now.